httpbp/middlewares_test.go

```diff
@@ -121,17 +135,17 @@ func TestInjectServerSpan(t *testing.T) {
 					t.Fatal("no binary annotations")
 				}
 				t.Logf("%#v", trace.BinaryAnnotations)
-				if c.err != nil {
-					hasError := false
-					for _, annotation := range trace.BinaryAnnotations {
-						if annotation.Key == "error" {
-							hasError = true
-						}
-					}
-					if !hasError {
-						t.Error("error binary annotation was not present.")
+				hasError := false
+				for _, annotation := range trace.BinaryAnnotations {
+					if annotation.Key == "error" {
+						hasError = true
 					}
 				}
+				if suppressor.Wrap(c.err) != nil && !hasError {
+					t.Error("error binary annotation was not present.")
+				} else if suppressor.Wrap(c.err) == nil && hasError {
+					t.Error("unexpected error binary annotation")
+				}
 			},
 		)
 	}
```

Comment:
can we get an Example of how to use it

Reply:
Did you want an example of using the `HTTPErrorSuppressor`? Since that's unexported now we probably don't need it?